### PR TITLE
Encryption of messages and files sending

### PR DESCRIPTION
**Pull request by gebsh**

### CODE REVIEW

File: encryptor/app.py [hunk 1]
```diff
@@ -18,10 +19,10 @@
 class MainWindow(QMainWindow):
     """Main window of the application."""
 
-    def __init__(self, port: int, keys_dir: str) -> None:
+    def __init__(self, port: int, keys_dir: Path) -> None:
         super().__init__()
 
-        self._server_thread = ServerThread(Address("127.0.0.1", port))
+        self._server_thread = ServerThread(Address("127.0.0.1", port), keys_dir)
```
Comment (gebsh) [inline encryptor/app.py:25]: Remove `keys_dir` from the server thread.

File: encryptor/network/client_thread.py [hunk 2]
```diff
@@ -100,6 +102,27 @@ def send_message(self, message: str) -> None:
...
+        file = open(filepath, "rb")
+
+        try:
+            if self._writer is not None:
+                data = encrypt(file.read(), self._mode, self._writer._endpoint_pubkey)
+
+                print(f"Sending a file {filepath} to the {self._writer.endpoint_addr}")
+                self._writer.write(Message.of(data, ContentType.FILE, self._mode, filepath.name))
+        finally:
+            file.close()
```
Comment (gebsh) [inline encryptor/network/client_thread.py]: `Path` class supports file operations directly, no need to use `open` here.

File: encryptor/network/message.py [hunk 3]
```diff
@@ -101,14 +89,18 @@ def __str__(self) -> str:
...
 
         return metaheader + headers + self.content
 
     @staticmethod
     def of(
-        content: bytes, content_type: ContentType, content_encoding: str = "utf-8"
+        content: bytes,
+        content_type: ContentType,
+        mode: Optional[EncryptionMode] = None,
+        filename: Optional[str] = None,
```
Comment (gebsh) [inline encryptor/network/message.py]: Move this to `**kwargs`.

File: encryptor/network/message.py [hunk 4]
```diff
@@ -118,10 +110,22 @@ def of(
...
+    def write_to_file(self, keys_dir) -> None:
+        """Writing bytes to file. Destination directory is based on argument"""
+
+        files_dir: Path = keys_dir / "files"
+        files_dir.mkdir(exist_ok=True)
+        file_path: Path = files_dir / self.headers.filename
+        if file_path.exists():
+            os.remove(file_path)
+        file_path.write_bytes(self.content)
+
```
Comment (gebsh) [inline encryptor/network/message.py]: This method should take `file_path` as an argument.

File: encryptor/app.py (hunk 1 above)
Comment (gebsh) [inline encryptor/app.py]: Remove `keys_dir` from the server thread.

File: encryptor/network/message.py [hunk 5]
```diff
@@ -146,30 +150,28 @@ def from_message(message: Message) -> "JSONMessageContent":
...
-        data["content_type"] = JSONContentType(data["content_type"])
-
         return JSONMessageContent(**data)
 
 
 class MessageReader:
     """Reads messages from an endpoint."""
 
-    def __init__(self, sock: socket.socket) -> None:
+    def __init__(self, sock: socket.socket, keys_dir: str) -> None:
```
Comment (gebsh) [inline encryptor/network/message.py]: Remove `keys_dir` from the server thread.

File: encryptor/network/message.py [hunk 6]
```diff
@@ -223,12 +230,22 @@ def read(self, content_type: Optional[ContentType] = None) -> Message:
...
+        if message_type == ContentType.FILE:
+            print(
+                f"New message from {self.endpoint_addr}: {message.headers.filename}\nSaving..."
+            )
+            message.write_to_file(keys_dir)
+        #            files_dir: Path = keys_dir / "files"
+        #            files_dir.mkdir(exist_ok=True)
+        #            file_path: Path = files_dir / message.headers.filename
+        #            file_path.write_bytes(message.content)
+
```
Comment (gebsh) [inline encryptor/network/message.py]: This should be somewhere else, probably in the main GUI thread.

File: encryptor/network/server_thread.py [hunk 7]
```diff
@@ -15,11 +18,12 @@ class ServerThread(QThread):
     new_message = pyqtSignal(Message)
     disconnect = pyqtSignal()
 
-    def __init__(self, addr: Address) -> None:
+    def __init__(self, addr: Address, keys_dir: Path) -> None:
```
Comment (gebsh) [inline encryptor/network/server_thread.py]: Remove `keys_dir` from the server thread.

File: encryptor/network/client_thread.py (hunk 2 above)
Comment (brtjank) [reply encryptor/network/client_thread.py]: done

File: encryptor/network/message.py (hunk 4 above)
Comment (brtjank) [reply encryptor/network/message.py]: done

File: encryptor/network/message.py (hunk 6 above)
Comment (brtjank) [reply encryptor/network/message.py]: done

File: encryptor/network/message.py (hunk 5 above)
Comment (brtjank) [reply encryptor/network/message.py]: done

File: encryptor/app.py (hunk 1 above)
Comment (brtjank) [reply encryptor/app.py]: done

File: encryptor/network/server_thread.py (hunk 7 above)
Comment (brtjank) [reply encryptor/network/server_thread.py]: done

**Pull request closed without merging**